hyperspy/roi.py

```diff
@@ -702,16 +702,16 @@ class SpanROI(BaseInteractiveROI):
     _ndim = 1
 
     def __init__(self, left, right):
-        super(SpanROI, self).__init__()
+        super().__init__()
         self._bounds_check = True   # Use reponsibly!
+        if left >= right:
+            raise ValueError("`left` must be smaller than `right`.")
```

Comment:
Could maybe add the values themselves to the error message. Sometimes function which raises the error isn't the one which the user called, so having the values themselves could be useful in that case. For example:

```python
raise ValueError("`left` ({0}) must be smaller than `right` ({1}).".format(left, right))
```